web/apps/labelstudio/src/components/Form/Elements/Select/Select.js

```diff
@@ -4,11 +4,30 @@ import { FormField } from '../../FormField';
 import { default as Label } from '../Label/Label';
 import './Select.styl';
 
-const Select = ({ label, className, options, validate, required, skip, labelProps, ghost, ...props }) => {
+const Select = ({ label, className, options, validate, required, skip, labelProps, groupProps, ghost, ...props }) => {
   const rootClass = cn('select');
   const initialValue = useMemo(() => props.value ?? "", [props.value]);
   const [value, setValue] = useState(initialValue);
 
+  let grouped = options.reduce((groupedOptions, option) => {
+    const key = option.group || 'NoGroup'; // fallback group for items without a group property
+    (groupedOptions[key] = groupedOptions[key] || []).push(option);
+    return groupedOptions;
+  }, {});
+
+  const renderOption = (option) => {
+    const value = option.value ?? option;
+    const label = option.label ?? value;
+    const disabled = option.disabled ?? false;
+    const hidden = option.hidden ?? false;
+
+    return (
+      <option key={value} value={value} disabled={disabled} hidden={hidden}>
+        {label}
+      </option>
+    );
+  };
```

Comment:
I would move this up to its own separate component definition to allow less logic to be defined within another.

```suggestion
  const SelectOption = ({ value, label, disabled = false, hidden = false, ...props }) => {
    return (
      <option value={value} disabled={disabled} hidden={hidden} {...props}>
        {label ?? value}
      </option>
    );
  };
```